pkg/tinydtls/contrib/sock_dtls.c

```diff
@@ -376,29 +442,81 @@ static ssize_t _copy_buffer(sock_dtls_t *sock, void *data, size_t max_len)
     if (buflen > max_len) {
         return -ENOBUFS;
     }
+#if SOCK_HAS_ASYNC
+    if (sock->buf_ctx != NULL) {
+        memcpy(data, buf, sock->buflen);
+        _check_more_chunks(sock->udp_sock, (void **)&buf, &sock->buf_ctx,
+                           &remote->ep);
+        if (sock->async_cb &&
+            /* is there a message in the sock's mbox? */
+            cib_avail(&sock->mbox.cib)) {
+            if (sock->buf) {
+                sock->async_cb(sock, SOCK_ASYNC_MSG_RECV,
+                               sock->async_cb_arg);
+            }
+            else {
+                sock->async_cb(sock, SOCK_ASYNC_CONN_RECV,
+                               sock->async_cb_arg);
+            }
+        }
+        return buflen;
+    }
+#else
+    (void)remote;
+#endif
     /* use `memmove()` as tinydtls reuses `data` to store decrypted data with an
      * offset in `buf`. This prevents problems with overlapping buffers. */
     memmove(data, buf, buflen);
     return buflen;
 }
 
+static ssize_t _complete_handshake(sock_dtls_t *sock,
+                                   sock_dtls_session_t *remote,
+                                   const session_t *session)
+{
+    memcpy(&remote->dtls_session, session, sizeof(remote->dtls_session));
+#ifdef SOCK_HAS_ASYNC
+    if (sock->async_cb) {
+        sock_async_flags_t flags = SOCK_ASYNC_CONN_RDY;
+
+        if (cib_avail(&sock->mbox.cib)) {
+            if (sock->buf) {
+                flags |= SOCK_ASYNC_MSG_RECV;
+            }
+            else {
+                flags |= SOCK_ASYNC_CONN_RECV;
```

Comment:
We already sent trigger the `CONN_RECV` callback in `_event`, do we need this again here?